### ISSUE TITLE
In built OSC in resonite doesn't work when using the Impressive mod.

### ISSUE THREAD

**Issue by ModernBalloonie** [RileyGuy/Impressive#7]
Self explanatory, Been wanting to use a hybrid of Impressive, but also using the in built osc for the extra params that the expression driver doesn't use. Found out osc doesn't work at all with the in built components when using the impressive mod.

**Comment by BlueCyro** [comment]
This might be because I'm providing my own version of Rug.Osc, try removing it from the rml_libs folder and see if that resolves the issue

**Comment by ModernBalloonie** [comment]
> This might be because I'm providing my own version of Rug.Osc, try removing it from the rml_libs folder and see if that resolves the issue

Just tried that, OSC is still broken.

**Comment by ModernBalloonie** [comment]
> This might be because I'm providing my own version of Rug.Osc, try removing it from the rml_libs folder and see if that resolves the issue

Just tried this, and I still have the issue. Interestingly enough after doing that the eye tracking still works, but I can't use the OSC Components. I even tried a different port.

**Comment by ModernBalloonie** [comment]
Confirmed using touch OSC as well. I set up a fader, it only works when Impressive is disabled.

**Comment by BlueCyro** [comment]
I can't replicate this on my machine. It appears to work just fine.

When making issues, please also be sure to test with only this single mod installed, make sure your modloader is up to date, and that you're on the latest version of Resonite. **Please** also provide logs when making issues on any mod as well, as these often aren't 'self-explanatory'.

Are you testing this in VR or Desktop? What version of Resonite? What version of the modloader? What version of Impressive?

**Comment by ModernBalloonie** [comment]
> I can't replicate this on my machine. It appears to work just fine.
> 
> When making issues, please also be sure to test with only this single mod installed, make sure your modloader is up to date, and that you're on the latest version of Resonite. **Please** also provide logs when making issues on any mod as well, as these often aren't 'self-explanatory'.
> 
> Are you testing this in VR or Desktop? What version of Resonite? What version of the modloader? What version of Impressive?

I just tried it in VR with only impressive installed. It seems the behaviour changed a little bit.

I can now get OSC signals on the in built components on other ports, but not 9000. Would there be any way to get the raw steam link face tracking values using the in game components while using impressive? Because i've tried and I can't get the values, I try typing things like "/sl/xrfb/facew/BrowLowererL" and they don't show up with anything.

[DESKTOP-QBGCSAQ - 2024.5.1.68 - 2024-05-01 17_51_15.log](https://github.com/BlueCyro/Impressive/files/15182705/DESKTOP-QBGCSAQ.-.2024.5.1.68.-.2024-05-01.17_51_15.log)
 Also here's the log.

**Comment by ModernBalloonie** [comment]
I just tried something else, I tried changing the port to 9015 so the face tracking wouldn't conflict with other things, and I changed impressive's and steam link's port to 9015. I installed the latest version of ResoniteModSettings to do so.

I went to use some things on port 9000 and they still didn't work. The face tracking, however did on port 9015. I then switched both steam link and impressive back to port 9000, and it seems like the face tracking completely broke when I changed it back. This only happened when I had resonite mod settings installed. I then tried switching the face tracking to 9015 again (both steam link and impressive) and the face tracking, still didn't work, neither were the OSC components.

[DESKTOP-QBGCSAQ - 2024.5.1.68 - 2024-05-01 18_25_47.log](https://github.com/BlueCyro/Impressive/files/15182868/DESKTOP-QBGCSAQ.-.2024.5.1.68.-.2024-05-01.18_25_47.log)

Here's the log.

Edit: On further investigation, it seems after restarting the game, stuff on port 9000 is fine if impressive is configured to use 9015 at game launch, but if it's changed in game, it seems to break ports it touches. Stuff on port 9000 works fine now, but that's after I restarted.
